examples/lock-app/efr32/src/LockManager.cpp

```diff
@@ -77,6 +99,64 @@ CHIP_ERROR LockManager::Init(chip::app::DataModel::Nullable<chip::app::Clusters:
     return CHIP_NO_ERROR;
 }
 
+bool LockManager::IsValidUserIndex(uint16_t userIndex)
+{
+    if (userIndex > kMaxUsers)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+bool LockManager::IsValidCredentialIndex(uint16_t credentialIndex, DlCredentialType type)
+{
+    // appclusters, 5.2.6.3.1: 0 is allowed index for Programming PIN credential only
+    if (DlCredentialType::kProgrammingPIN == type)
+    {
+        return (0 == credentialIndex);
+    }
+
+    if (credentialIndex > kMaxCredentialsPerUser)
```

Comment:
do we need to check for 0 as being invalid?

below it says "check programing pin" ... it sems already checked above. Here we are guaranteed type != kProgrammingPIN.

Reply:
Just realized this is 0-indexed ... so likely off by one ... or is this method one-indexed? Maybe methods expecting one-index should have `OneIndexed` as suffix or something similar.

Reply:
This method receives a zeroIndexed index. Programming pin is the only credential type allowed to have "0" as an index. I don't really like this, but that's why I have 0 == credentialIndex in the programming pin case

I removed the extra comment.